plugin/pkg/scheduler/schedulercache/cache.go

```diff
@@ -165,18 +165,17 @@ func (cache *schedulerCache) ForgetPod(pod *v1.Pod) error {
 		return fmt.Errorf("pod %v state was assumed on a different node", key)
 	}
 
-	switch {
 	// Only assumed pod can be forgotten.
-	case ok && cache.assumedPods[key]:
-		err := cache.removePod(pod)
-		if err != nil {
-			return err
-		}
-		delete(cache.assumedPods, key)
-		delete(cache.podStates, key)
-	default:
+	if !ok || !cache.assumedPods[key] {
```

Comment:
Its logic seems a bit complex -- only one case, and default returns an error. It makes me wonder when it would return an error, when it would go on.